usr/share/rear/build/GNU/Linux/100_copy_as_is.sh

```diff
@@ -1,53 +1,72 @@
 # 400_copy_as_is.sh
 #
-# copy files and directories that should be copied over as-is to the rescue
-# systems. Checks also for library dependencies of executables and adds
-# them to the LIBS list, if they are not included in the copied files.
+# Copy files and directories that should be copied as-is into the recovery system.
+# Check also for library dependencies of executables in all the copied files and
+# add them to the LIBS list if they are not yet included in the copied files.
 
 LogPrint "Copying files and directories"
 Log "Files being copied: ${COPY_AS_IS[@]}"
 Log "Files being excluded: ${COPY_AS_IS_EXCLUDE[@]}"
 
-for f in "${COPY_AS_IS_EXCLUDE[@]}" ; do echo "$f" ; done >$TMP_DIR/copy-as-is-exclude
-tar -v -X $TMP_DIR/copy-as-is-exclude \
-	-P -C / -c "${COPY_AS_IS[@]}" 2>$TMP_DIR/copy-as-is-filelist | \
-	tar $v -C $ROOTFS_DIR/ -x >/dev/null
-StopIfError "Could not copy files and directories"
-Log "Finished copying COPY_AS_IS"
-
-# fix ReaR directory if running from checkout
-if [[ "$REAR_DIR_PREFIX" ]] ; then
-    for dir in /usr/share/rear /var/lib/rear ; do
-        ln $v -sf $REAR_DIR_PREFIX$dir $ROOTFS_DIR$dir>/dev/null
+local copy_as_is_filelist_file="$TMP_DIR/copy-as-is-filelist"
+local copy_as_is_exclude_file="$TMP_DIR/copy-as-is-exclude"
+
+# Build the list of files and directories that are excluded from being copied:
+local excluded_file=""
+for excluded_file in "${COPY_AS_IS_EXCLUDE[@]}" ; do
+    echo "$excluded_file"
+done >$copy_as_is_exclude_file
+
+# Copy files and directories as-is into the recovery system except the excluded ones and
+# remember what files and directories were actually copied in a copy_as_is_filelist_file
+# which is the reason that the first 'tar' must be run in verbose mode:
+if ! tar -v -X $copy_as_is_exclude_file -P -C / -c "${COPY_AS_IS[@]}" 2>$copy_as_is_filelist_file | tar $v -C $ROOTFS_DIR/ -x 1>/dev/null ; then
+    Error "Failed to copy files and directories in COPY_AS_IS minus COPY_AS_IS_EXCLUDE"
+fi
+Log "Finished copying files and directories in COPY_AS_IS minus COPY_AS_IS_EXCLUDE"
+
+# Build an array of the actual regular files that are executable in all the copied files:
+local copy_as_is_executables=()
+local copy_as_is_file=""
+while read -r copy_as_is_file ; do
+    # Skip non-regular files (like directories and device files):
+    test -f "$copy_as_is_file" || continue
+    # Skip symbolic links (only care about symbolic link targets):
+    test -L "$copy_as_is_file" && continue
```

Comment:
Hi @jsmeix , do you know why to skip symbolic links? The previous versions did not have this (the test was `if [[ ! -d "$REPLY" && -x "$REPLY" ]]; then`, which includes symlinks).

I believe skipping symlinks is wrong: if the symlink is dangling in the recovery system, the target will be eventually copied by `build/default/490_fix_broken_links.sh`, but without libraries, and verification of the recovery system will fail and abort `rear mkrescue`.
(If the symlink is not dangling, there will be duplicates in the library list if we don't skip it, but I suppose that's not a big problem?)

Reply:
Removing this line does not seem to have any ill effect in a quick test

Reply:
@pcahyna 
could you please make a new GitHub issue for it
where you explain how things went wrong for you
so I could better understand the whole thing
and we could better discuss it
(I do no longer trust comments in pull requests).

I don't remember why I skip symlinks there.
The initial comment of this code part
```

# Build an array of the actual regular files that are executable in all the copied files:
```
also only tells that symlinks are not considered
but it does not tell why.

I assume your failing case is a rather special case
because I committed that change on Sep 29 2017 as
https://github.com/rear/rear/commit/3973399cbf79c51fb9901ade8039a34740418ef8
and I merged its pull request on Oct 4 2017
so since 6 years it had worked without issues
(at least without reported issues as far as I know)
which is why I like to fully understand the root cause
behind your current issue why it now fails for you
before I "just change" any symlink behaviour in ReaR
because in practice symlinks are a hell of endless pain
(not only in ReaR but everywhere).

For the (not so) fun of it:
In the end symlinks are just one implementation of
RFC 1925 item 6a
"It is always possible to add another level of indirection"
which is (in my experience) a root cause behind 90% of issues.
In my experience another root cause behind 90% of issues
is RFC 1925 item 5 "aglutenate [sic] multiple separate
problems into a single complex interdependent solution"
so about 81% of issues have RFC 1925 items 5 and 6a
as root causes :-(

Reply:
@jsmeix , #3064 . It also explains why this problem does not arise very often.

Reply:
After quick reading https://github.com/rear/rear/issues/3064#issue-1974321989
I can now imagine even a possible reason why COPY_AS_IS
does not care about symlinks like PROGS does:
Because COPY_AS_IS means "copy as is but nothing more"?
This is only an offhanded guess.